apps/etterlatte-saksbehandling-ui/client/src/components/behandling/soeknadsoversikt/viderefoere-opphoer/ViderefoereOpphoerVurdering.tsx

```diff
@@ -1,260 +1,150 @@
-      const vilkaartype = vilkaar ? finnVilkaartypeFraTittel(vilkaartyperResult.data, vilkaar)?.name : undefined
-      return setViderefoertOpphoer(
-        { skalViderefoere, behandlingId, begrunnelse, vilkaarType: vilkaartype, opphoerstidspunkt },
-        (viderefoertOpphoer) => {
-          dispatch(oppdaterViderefoertOpphoer(viderefoertOpphoer))
-          dispatch(oppdaterBehandlingsstatus(IBehandlingStatus.OPPRETTET))
-          onSuccess?.()
-        }
-      )
-    }
-  }
-
-  const reset = (onSuccess?: () => void) => {
-    resetToInitial()
-    resetSlettToInitial()
-    resetHentVilkaarToInitial()
-    setSkalViderefoere(viderefoertOpphoer?.skalViderefoere)
-    setVilkaar(viderefoertOpphoer?.vilkaar)
-    setOpphoerstidspunkt(viderefoertOpphoer?.dato ? new Date(viderefoertOpphoer.dato) : null)
-    setVilkaarError('')
-    setBegrunnelse(viderefoertOpphoer?.begrunnelse || '')
-    setVurdert(viderefoertOpphoer !== null)
-    onSuccess?.()
-  }
-
-  const finnVilkaartypeFraTittel = (vilkaar: Vilkaartyper, tittel: string) => {
-    return vilkaar.typer.find((p) => p.tittel === tittel)
-  }
-
-  const { monthpickerProps, inputProps } = useMonthpicker({
-    fromDate: virkningstidspunkt ?? opphoerstidspunkt,
-    toDate: addMonths(new Date(), 6),
-    onMonthChange: (date: Date) => setOpphoerstidspunkt(date),
-    inputFormat: 'dd.MM.yyyy',
-    onValidate: (val) => {
-      if (val.isBefore || val.isAfter) setVilkaarError('Opphørstidspunkt er ikke gyldig')
-      else setVilkaarError('')
+  const [lagreViderefoertOpphoerStatus, lagreViderefoertOpphoerRequest] = useApiCall(lagreViderefoertOpphoer)
+  const { handleSubmit, control, register, watch } = useForm<ViderefoertOpphoerVurderingData>({
+    shouldUnregister: true,
+    defaultValues: {
+      skalViderefoere: viderefoertOpphoer ? viderefoertOpphoer.skalViderefoere : undefined,
+      dato: viderefoertOpphoer ? viderefoertOpphoer.dato : undefined,
+      vilkaar: viderefoertOpphoer ? (finnTittelFraVilkaartype(vilkaarTyper, viderefoertOpphoer.vilkaar) ?? '') : '',
+      begrunnelse: viderefoertOpphoer ? viderefoertOpphoer.begrunnelse : undefined,
     },
-    defaultSelected: opphoerstidspunkt ?? undefined,
-  } as UseMonthPickerOptions)
+  })
 
-  function haandterViderefoertOpphoer(event: JaNei) {
-    const jaNeiElement = JaNei[event as JaNei]
-    setSkalViderefoere(jaNeiElement)
-    setVilkaarError('')
-    if (jaNeiElement === JaNei.NEI) {
-      setOpphoerstidspunkt(null)
-      setVilkaar(undefined)
-    }
+  const lagreViderefoerOpphoer = (data: ViderefoertOpphoerVurderingData) => {
+    lagreViderefoertOpphoerRequest(
+      {
+        behandlingId,
+        skalViderefoere: data.skalViderefoere,
+        vilkaarType: data.vilkaar != '' ? finnVilkaartypeFraTittel(vilkaarTyper, data.vilkaar)?.name : undefined,
+        begrunnelse: data.begrunnelse != '' ? data.begrunnelse : undefined,
```

Comment:
begge kan forenkles til formatet `vilkaarType: !!data.vilkaar && finnVilkaartypeFraTittel(vilkaarTyper, data.vilkaar)?.name`

Reply:
Her er det ikke rett frem i og med at vilkaar må være tom streng for at ikke den comboboksen skal feile. Tom streng er at ingen vilkår er valgt.